gsoc/common/utils/irc.py

```diff
@@ -0,0 +1,54 @@
+import json
+from random import randint
+
+from fredirc import BaseIRCHandler
+from fredirc import Err
+from fredirc import IRCClient
+
+import gsoc.settings as config
+
+class ModIRCClient(IRCClient):
+
+    def __init__(self, handler, nick, server, messages):
+        IRCClient.__init__(self, handler, nick, server)
+        self.messages = messages
+
+class CommandBot(BaseIRCHandler):
+
+    def handle_register(self):
+        for message in self.client.messages:
+            commands = process_message(message)
+            for command in commands:
+                self.client.send_private_message(config.RECEIVER, command)
+        self.client.quit()
+
+    def handle_disconnect(self):
+        self.client.terminate()
+
+    def handle_error(self, num, **params):
+        if num == Err.NICKNAMEINUSE:
+            new_nick = params['nick'] + str(randint(1, 9))
+            self.client.register(nick = new_nick)
+
+def process_message(message):
```

Comment:
This looks like the code would work, but where is it used?

Reply:
I didn't get you, are you talking about the nickname change if nickname is already in use?